script/tool/lib/src/version_check_command.dart

```diff
@@ -140,11 +155,26 @@ class VersionCheckCommand extends PackageLoopingCommand {
 
     final List<String> errors = <String>[];
 
-    if (!await _hasValidVersionChange(package, pubspec: pubspec)) {
-      errors.add('Disallowed version change.');
+    bool versionChanged = false;
+    final _CurrentVersionState versionState =
+        await _getVersionState(package, pubspec: pubspec);
+    switch (versionState) {
+      case _CurrentVersionState.unchanged:
+        break;
```

Comment:
nit: I'd put versionChanged = false here since setting it above isn't clearly associated with the unchanged branch.

Reply:
Changed to no initialization of the variable so that the compiler forces setting it in each branch, for better clarity/future-proofing.